Makefile

```diff
@@ -1,8 +1,25 @@
-BIN := drone-google-cloudstorage
-IMG ?= plugins/$(BIN)
+.PHONY: clean deps test build docker
 
-docker: $(BIN) Dockerfile
-	docker build --rm -t $(IMG) .
+export GOOS ?= linux
+export GOARCH ?= amd64
+export CGO_ENABLED ?= 0
 
-$(BIN): $(wildcard *.go)
-	GOOS=linux GOARCH=amd64 go build
+CI_BUILD_NUMBER ?= 0
+
+LDFLAGS += -X "main.buildDate=$(shell date -u '+%Y-%m-%d %H:%M:%S %Z')"
+LDFLAGS += -X "main.build=$(CI_BUILD_NUMBER)"
+
+clean:
+	go clean -i ./...
+
+deps:
+	go get -t ./...
+
+test:
+	go test -cover ./...
+
+build:
+	go build -ldflags '-s -w $(LDFLAGS)'
+
+docker:
+	docker build --rm=true -t plugins/drone-google-cloudstorage .
```

Comment:
A better way to do his (esp. given the defined `.PHONY` which suggest a certain usage pattern) is something like:

```
docker: .docker
.docker: drone-google-cloudstorage Dockerfile
	docker build --rm=true -t plugins/drone-google-cloudstorage .
	touch .docker
```

That way the image will be built only when source code or Dockerfile have changed since the last build.